cmd/acikubectl/cmd/root.go

```diff
@@ -76,9 +80,23 @@ func initClient() (kubernetes.Interface, error) {
 
 	if kubeconfig != "" {
 		// use kubeconfig file from command line
-		restconfig, err = clientcmd.BuildConfigFromFlags("", kubeconfig)
-		if err != nil {
-			return nil, err
+		// the context is only useful with a kubeconfig
+		if context == "" {
+			restconfig, err = clientcmd.BuildConfigFromFlags("", kubeconfig)
+			if err != nil {
+				return nil, err
+			}
+		} else {
+			restconfig, err = clientcmd.NewNonInteractiveDeferredLoadingClientConfig(
+				&clientcmd.ClientConfigLoadingRules{
+					ExplicitPath: kubeconfig,
+				},
+				&clientcmd.ConfigOverrides{
+					CurrentContext: context,
+				}).ClientConfig()
+			if err != nil {
+				return nil, err
+			}
```

Comment:
It's a bit silly to have both calls to BuildConfigFromFlags and the explicit NewNonInteractiveDeferredLoadingClientConfig, given that BuildConfigFromFlags just calls NewNonInteractiveDeferredLoadingClientConfig and passes in the kubeconfig.  You can simplify this all to just the "else" branch above.

Reply:
(See https://github.com/kubernetes/client-go/blob/master/tools/clientcmd/client_config.go#L551)

Reply:
What happens if I pass a kubeconfig without a context, though?  I'd assumed that the Kubernetes client would then interpret the default empty string as the context and error out.  I didn't see a way around needing to pass `&clientcmd.ConfigOverrides` in one case and nothing in the other case.

Reply:
default constructed value of context is going to be ""